packages/registry/src/index.ts

```diff
@@ -181,6 +185,214 @@ class Config {
+  }): void {
+    const { name, component, predicates, slot } = options;
+
+    if (!component) {
+      throw new Error('No component provided');
+    }
+    if (!predicates) {
+      // Test if there's already one registered, we only support one
+      const hasRegisteredNoPredicatesComponent = this._data.slots?.[
+        slot
+      ]?.data?.[name]?.find(({ predicates }) => !predicates);
+      if (hasRegisteredNoPredicatesComponent) {
+        throw new Error(
+          `There is already registered a component ${name} for the slot ${slot}. You can only register one slot component with no predicates per slot.`,
+        );
+      }
+    }
+
+    let currentSlot = this._data.slots[slot];
+    if (!currentSlot) {
+      this._data.slots[slot] = {
+        slots: [],
+        data: {},
+      };
+      currentSlot = this._data.slots[slot];
+    }
+    if (!currentSlot.data[name]) {
+      currentSlot.data[name] = [];
+    }
+
+    const currentSlotComponents = currentSlot.data[name];
+    if (!currentSlot.slots.includes(name)) {
+      currentSlot.slots.push(name);
+    }
+    const slotComponentData = {
+      component,
+      predicates,
+    };
+
+    // Try to set a displayName (useful for React dev tools) for the registered component
+    // Only if it's a function and it's not set previously
+    try {
+      const displayName = slotComponentData.component.displayName;
+
+      if (!displayName && typeof slotComponentData?.component === 'function') {
+        slotComponentData.component.displayName = name;
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `Not setting the slot component displayName because ${error}`,
+      );
+    }
+
+    currentSlotComponents.push(slotComponentData);
+  }
+
+  getSlotComponent(slot: string, name: string) {
+    const currentSlot = this._data.slots[slot];
+    if (!slot || !currentSlot) {
+      throw new Error(`No slot ${slot} found`);
+    }
+    const currentSlotComponents = currentSlot.data[name];
+    if (!currentSlotComponents) {
+      throw new Error(`No slot component ${name} in slot ${slot} found`);
+    }
+    return currentSlotComponents;
+  }
+
+  getSlotComponents(slot: string) {
+    const currentSlot = this._data.slots[slot];
+    if (!slot || !currentSlot) {
+      throw new Error(`No slot ${slot} found`);
+    }
+    return currentSlot.slots;
+  }
+
+  reorderSlotComponent({
+    slot,
+    name,
+    position,
+    action,
+    target,
+  }: {
+    slot: string;
+    name: string;
+    position?: number;
+    action?: 'after' | 'before' | 'first' | 'last';
+    target?: string;
+  }) {
+    if (!position && !action) {
+      throw new Error(`At least a position or action is required as argument`);
+    }
+    if (position && target) {
+      throw new Error(
+        `You should provide only one of position or action as arguments`,
+      );
+    }
+    if (action && action !== 'first' && action !== 'last' && !target) {
```

Comment:
This is aligns better with the error message. Otherwise I have to check other values not in the enum. Also by changing the previous error condition, we can remove `action &&`.

```suggestion
    if ((action == 'after' || action == 'before') && !target) {
```

Reply:
Pushed, I run the tests to make sure.